src/DocumentDB.ChangeFeedProcessor.IntegrationTests/EstimatorTests.cs

```diff
@@ -126,5 +126,72 @@ public EstimatorTests(IntegrationTestFixture fixture) : base(fixture, typeof(Est
                 }
             }
         }
+
+        [Fact]
+        public async Task WhenNoLeasesReturn1()
+        {
+            var hostName = Guid.NewGuid().ToString();
+
+            var host = new ChangeFeedEventHost(
+                hostName,
+                this.ClassData.monitoredCollectionInfo,
+                this.LeaseCollectionInfo,
+                new ChangeFeedOptions { StartFromBeginning = false },
+                new ChangeFeedHostOptions());
+
+            // Verify that 1 is returned on an uninitialized collection
+            long estimation = await host.GetEstimatedRemainingWork();
+            Assert.Equal(1, estimation);
+        }
+
+        [Fact]
+        public async Task WhenUninitializedLeasesStartFromBeginning()
+        {
+            int documentCount = 1;
+            int partitionCount = await IntegrationTestsHelper.GetPartitionCount(this.ClassData.monitoredCollectionInfo);
+            int openedCount = 0, processedCount = 0;
+            var allObserversStarted = new ManualResetEvent(false);
+            var allDocsProcessed = new ManualResetEvent(false);
+
+            var observerFactory = new TestObserverFactory(
+                context =>
+                {
+                    int newCount = Interlocked.Increment(ref openedCount);
+                    if (newCount == partitionCount) allObserversStarted.Set();
+                    return Task.CompletedTask;
+                },
+                null,
+                (ChangeFeedObserverContext context, IReadOnlyList<Document> docs) =>
+                {
+                    int newCount = Interlocked.Add(ref processedCount, docs.Count);
+                    if (newCount == documentCount) allDocsProcessed.Set();
+                    return Task.CompletedTask;
+                });
+
+            var hostName = Guid.NewGuid().ToString();
+
+            var host = new ChangeFeedEventHost(
+                hostName,
+                this.ClassData.monitoredCollectionInfo,
+                this.LeaseCollectionInfo,
+                new ChangeFeedOptions { StartFromBeginning = false },
+                new ChangeFeedHostOptions());
+
+            // Initialize leases
```

Comment:
Please clarify what this step is doing and why it's needed.

Reply:
I will add comments. This will initialize the host and generate leases. The leases will have ContinuationToken null initially.